include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -866,7 +866,9 @@ class alignment_file_output_format<format_bam> : alignment_file_output_format<fo
                     {
                         if (!std::ranges::empty(id_source)) // otherwise default will remain (-1)
                         {
-                            auto id_it = header.ref_dict.find(id_source);
+                            using ref_id_t = std::remove_reference_t<decltype(header.ref_ids()[0])>;
+                            auto key = id_source | std::ranges::to<ref_id_t>;
+                            auto id_it = header.ref_dict.find(key);
```

Comment:
change the key-type of the map to be ranges::subrange and decay contiguous iterators to pointers. 

Add a static_assert that the respective types are convertible to the key_type.